spec.md

```diff
@@ -413,6 +419,22 @@ message CreateVolumeResponse {
   }
 }
 
+message Principal {
+  // describes attributes for defining the principal identity of 
+  // a volume.
+
+  // This field is OPTIONAL. If specified it contains the CO namespace
+  // identity in which the the following identity information (user /
+  // group) exists.
+  string namespace = 1;
```

Comment:
This sort of access mgmt could be done with a third party IAM/RBAC  solution. The users and groups and roles being defined there.  Is that also an assumption here? The SP could be using these to manage access to its storage already. The CO need not be the one managing users/groups/roles.

I think the spec also needs to call out and define what is user/group/role and the fact that both CO and the SP can be defining these. In which case how is that proposed to be handled.

Please also define the behavior for "no authorization" - does it mean all access or no access.